autogen/main/main.tf.tmpl

```diff
@@ -111,8 +111,12 @@ locals {
     security_group = var.authenticator_security_group
   }]
 
-  cluster_node_metadata_config = var.node_metadata == "UNSPECIFIED" ? [] : [{
-    node_metadata = var.node_metadata
+  // legacy mappings https://github.com/hashicorp/terraform-provider-google/pull/10238
+  old_node_metadata_config_mapping = { GKE_METADATA_SERVER = "GKE_METADATA", GCE_METADATA = "EXPOSE" }
+  mapped_node_metadata_config      = lookup(local.old_node_metadata_config_mapping, var.node_metadata, var.node_metadata)
```

Comment:
This local seems like an unnecessary layer of indirection - could just have the lookup on line 119.

Reply:
good catch, missed that UNSPECIFIED was common